src/java.net.http/share/classes/jdk/internal/net/http/Http2Connection.java

```diff
@@ -700,8 +700,7 @@ void shutdown(Throwable t) {
         Throwable initialCause = this.cause;
         if (initialCause == null) this.cause = t;
         client2.deleteConnection(this);
-        List<Stream<?>> c = new ArrayList<>(streams.values());
-        for (Stream<?> s : c) {
+        for (Stream<?> s : streams.values()) {
             try {
```

Comment:
To address some of the feedback given, the streams  are now directly addressed instead of making a copy.  ConcurrentMap has also been used to initialise streams instead of Map. Seeking feedback/opinions on this if possible